icu4c/source/common/unicode/unistr.h

```diff
@@ -2925,6 +3022,34 @@ class U_COMMON_API UnicodeString : public Replaceable
    */
   const char16_t *getTerminatedBuffer();
 
+#ifndef U_HIDE_DRAFT_API
+  /**
+   * Converts to a std::u16string_view.
+   *
+   * @return a string view of the contents of this string
+   * @draft ICU 76
+   */
+  inline operator std::u16string_view() const {
+    return { getBuffer(), (std::u16string_view::size_type)length() };
+  }
+
+#if U_SIZEOF_WCHAR_T==2 || defined(U_IN_DOXYGEN)
+  /**
+   * Converts to a std::wstring_view.
```

Comment:
@eggrobin heads-up this is still here for Windows...

Reply:
With my liaison officer’s hat on, this one makes me nervous, because it will be unsafe when we have a good solution on the C++ side.
With my Windows developer’s hat on, I need _something_.

I think it would be safer to have an actual member function here, that way once C++ figures things out, it can be made to return an RAII object that transmutes back after use.

But with my ICU-TC delegate’s hat on, I lean on keeping whatever we do on this particular conversion draft even as we undraft the rest, until we can see more clearly what will happen on the C++ side. Which means we might as well do this for now.

Reply:
Adding a note about keeping it draft.